src/Native/System.Native/pal_networkchange.cpp

```diff
@@ -52,7 +52,8 @@ extern "C" NetworkChangeKind SystemNative_ReadSingleEvent(int32_t sock)
     iovec iov = {buffer, sizeof(buffer)};
     sockaddr_nl sanl;
     msghdr msg = {reinterpret_cast<void*>(&sanl), sizeof(sockaddr_nl), &iov, 1, NULL, 0, 0};
-    ssize_t len = recvmsg(sock, &msg, 0);
+    ssize_t len;
+    while (CheckInterrupted(len = recvmsg(sock, &msg, 0)));
```

Comment:
> I'll leave it up to you what you think should be done here (after this PR). 

I honestly have no preference either way.  @pgavlin - if you feel strongly, can you open an issue and assign it to me?  I'll change it across all the shims.